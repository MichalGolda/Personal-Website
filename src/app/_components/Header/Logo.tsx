import Image from "next/image";

export const Logo = () => {
  return (
    <a href="#top">
      <Image src="/logo.svg" alt="logo" width={151} height={29} />
    </a>
  );
};
