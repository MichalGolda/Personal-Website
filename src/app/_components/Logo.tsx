import Image from "next/image";

const Logo = () => {
  return (
    <a href="#top">
      <Image src="/logo.svg" alt="logo" width={151} height={29} />
    </a>
  );
};

export default Logo;
