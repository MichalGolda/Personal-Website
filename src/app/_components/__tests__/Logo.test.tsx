import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import Logo from "../Logo";

describe("Logo component", () => {
  it("renders without errors", () => {
    const { container } = render(<Logo />);

    const link = container.firstChild;
    const image = link?.firstChild;

    expect(link).toHaveAttribute("href", "#top");

    expect(image).toHaveAttribute("src", "/logo.svg");
    expect(image).toHaveAttribute("alt", "logo");
    expect(image).toHaveAttribute("width", "151");
    expect(image).toHaveAttribute("height", "29");
  });

  it("renders unchanged", () => {
    const { container } = render(<Logo />);

    expect(container).toMatchSnapshot();
  });
});
