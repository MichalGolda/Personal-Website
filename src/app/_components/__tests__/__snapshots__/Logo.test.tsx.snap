// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo component renders unchanged 1`] = `
<div>
  <a
    href="#top"
  >
    <img
      alt="logo"
      data-nimg="1"
      decoding="async"
      height="29"
      loading="lazy"
      src="/logo.svg"
      style="color: transparent;"
      width="151"
    />
  </a>
</div>
`;
